### PR TITLE
improve exception handling to clean up sentry issue list

### PR DESCRIPTION
**Pull request by lsmith77**
fixes NLP_API-2F

* do not throw an exception if lang could not be determined
* raise exception of LT returns a 500

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -216,20 +218,15 @@ def get_root():
     return RedirectResponse(url="/form", status_code=301)
 
 
-@app.get("/exception", response_model=ResultsOut)
+@app.post("/exception")
```
Comment (lsmith77) [inline app/main.py:221]: this end-point is just for testing and I adjusted to mimic more closely `/check`

File: app/models.py [hunk 2]
```diff
@@ -313,12 +313,27 @@ def getGenderedRolesFormatInclusive(alternative, german_gender_ending):
...
 class ResultsOut(BaseModel):
     results: List[ResultOut]
     language: str
 
     def factory(results, lang):
-        return ResultsOut(results, lang.lang)
+        if lang != None:
+            lang = lang.lang
+
+        return ResultsOut(results, lang)
```
Comment (Szarad) [inline app/models.py:336]: Should we call ResultsOut when lang is None?

File: app/models.py [hunk 3]
```diff
@@ -313,12 +313,27 @@ def getGenderedRolesFormatInclusive(alternative, german_gender_ending):
...
+        object.__setattr__(self, "details", details)
+
+
 class ResultsOut(BaseModel):
     results: List[ResultOut]
     language: str
 
     def factory(results, lang):
-        return ResultsOut(results, lang.lang)
+        if lang != None:
```
Comment (lsmith77) [inline app/models.py:333]: this is technically not needed but just to be sure

**Comment by Szarad** [comment]
ok, then looks good to me

**Review by Szarad — APPROVED** [review]

File: app/models.py (hunk 2 above)
Comment (lsmith77) [reply app/models.py:336]: no .. technically this should never happen

**Pull request merged**